aws/app/lambda.tf

```diff
@@ -613,3 +613,47 @@ resource "aws_cloudwatch_log_group" "nagware" {
   kms_key_id        = var.kms_key_cloudwatch_arn
   retention_in_days = 90
 }
+
+#
+# Vault data integrity check
+#
+
+data "archive_file" "vault_data_integrity_check_main" {
+  type        = "zip"
+  source_file = "lambda/vault_data_integrity_check/vault_data_integrity_check.js"
+  output_path = "/tmp/vault_data_integrity_check_main.zip"
+}
+
+resource "aws_lambda_function" "vault_data_integrity_check" {
+  filename      = "/tmp/vault_data_integrity_check_main.zip"
+  function_name = "VaultDataIntegrityCheck"
+  role          = aws_iam_role.lambda.arn
+  handler       = "vault_data_integrity_check.handler"
+  timeout       = 60
+
+  source_code_hash = data.archive_file.vault_data_integrity_check_main.output_base64sha256
+
+  runtime = "nodejs18.x"
+
+  tracing_config {
+    mode = "PassThrough"
+  }
+
+  tags = {
+    (var.billing_tag_key) = var.billing_tag_value
+    Terraform             = true
+  }
+}
+
+resource "aws_lambda_event_source_mapping" "vault_updated_item_stream" {
+  event_source_arn                   = var.dynamodb_vault_stream_arn
+  function_name                      = aws_lambda_function.vault_data_integrity_check.arn
```

Comment:
As mentionned in another comment filtering could be added here instead to reduce the load on the Lambda by only sending requests you want to check.
https://registry.terraform.io/providers/hashicorp/aws/3.76.1/docs/resources/lambda_event_source_mapping#filter_criteria

Reply:
Yeah that is correct. I can add it. I did not do it in the first place because I thought there may be a scenario where we would want to check data that has been deleted but then forgot about it.